services/sns.rb

```diff
@@ -73,6 +73,16 @@ def publish_message(msg)
     raise_error e.message
   end
 
+  def json_message_generator_for(msg)
+    json = {
+      :default => msg.to_json
+    }
+
+    json.tap do
```

Comment:
This seems fancy for not much gain. What does this section look like without using `tap`?